File: solutions/julia/cars-assemble/3/cars-assemble.jl
const base_prod_rate = 221

function success_rate(speed)
    denom = if speed <= 0
        0
    elseif 1 <= speed <= 4
        100
    elseif 5 <= speed <= 8
        90
    elseif speed == 9
        80
    else
        77
    end

    return denom / 100
end

production_rate_per_hour(speed) = base_prod_rate * speed * success_rate(speed)

working_items_per_minute(speed) = return Int(floor(production_rate_per_hour(speed) / 60))

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private static final int BASE_PRODUCTION_RATE = 221;

    public double productionRatePerHour(int speed) {
        if (speed <= 0) {
            return 0;
        }

        double actualProductionRate;
        
        if (speed < 5) {
            actualProductionRate = BASE_PRODUCTION_RATE;
        } else if (speed < 9) {
            actualProductionRate = BASE_PRODUCTION_RATE * 0.9;
        } else if ( speed == 9) {
            actualProductionRate = BASE_PRODUCTION_RATE * 0.8;
        } else {
            actualProductionRate = BASE_PRODUCTION_RATE * 0.77;
        }

        return speed * actualProductionRate;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) this.productionRatePerHour(speed) / 60;
    }
}
